### DISCUSSION TITLE
[gatsby-remark-images]: What's the purpose of generating 2x and 3x sizes of thumbnails?

### DISCUSSION THREAD

**Discussion by bytrangle** [gatsbyjs/gatsby#33254] [Help]
Sorry if this is obvious to others, but I don't understand.

By default, Gatsby generates 0.25x, 0.5x, 1x, 1.5x, 2x, and 3x sizes of thumbnails.

This is an example of responsive image handling by `gatsby-remark-images`:

![2021-09-20 17_00_14-gatsby-remark-images](https://user-images.githubusercontent.com/28924121/133985493-fdb93e08-b493-430f-a2b6-85d197f9d751.png)

```
<img class="gatsby-resp-image-image css-0" alt="The workflow for how your code gets from your computer to your users. Extended description below." title="The workflow for how your code gets from your computer to your users. Extended description below." src="/static/0fd27b745c1de708f034eaf97c4416e0/321ea/deployment-workflow.png" srcset="/static/0fd27b745c1de708f034eaf97c4416e0/1c154/deployment-workflow.png 197w,/static/0fd27b745c1de708f034eaf97c4416e0/d72ec/deployment-workflow.png 393w,/static/0fd27b745c1de708f034eaf97c4416e0/321ea/deployment-workflow.png 786w,/static/0fd27b745c1de708f034eaf97c4416e0/60b3a/deployment-workflow.png 1179w,/static/0fd27b745c1de708f034eaf97c4416e0/5df5d/deployment-workflow.png 1572w,/static/0fd27b745c1de708f034eaf97c4416e0/d61c2/deployment-workflow.png 1800w" sizes="(max-width: 786px) 100vw, 786px" style="width: 100%; height: 100%; margin: 0px; vertical-align: middle; position: absolute; top: 0px; left: 0px; opacity: 1; color: inherit; box-shadow: rgb(255, 255, 255) 0px 0px 0px 400px inset;" loading="lazy">
```

The width of images for browser to choose from are: 197, 393, 786, 1179, 1572 and 1800.

The first five values correspond to 0.25x, 0.5x, 1x, 1.5x and 2x of the value 786. I don't know where the number 1800 comes from. Is it the intrinsic width of the uploaded image?

The `sizes` attribute dictates that on viewports 786px wide and below, the image will take up 100% of the viewport. In other cases, the image will occupy exactly 786px.

This means that the image to be loaded by the browser will never be larger than 786px wide.

So what is the point of giving big images like 1179, 1572 and 1800 to the browser? If I understand correctly, they will never get used.

**Comment by KyleAMathews** [comment]
Higher density screens can display images at 2x their "native" resolution. The 1800 image is I believe the original image width. I think we did remove that from gatsby-plugin-image but gatsby-remark-image might not have gotten the same change.

**Reply by bytrangle** [reply]
Yes, I forget to add that 1800 is the original image's width. So 2x, 3x images are needed for high-density screens, but the original image width is redundant?
